host/src/containers/GameSessionContainer.tsx

```diff
@@ -104,20 +112,66 @@ const GameSessionContainer = () => {
     }
   }, [isModalOpen]);
 
+  useEffect(() => { //headerGame timer, 1 second interval, update localstorage in case page resets
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+        if (headerGameCurrentTime > 0) {
+          setHeaderGameCurrentTime(headerGameCurrentTime - 1);
+          localStorage.setItem('currentGameTimeStor', headerGameCurrentTime-1);
+        }
+        else
+          setGameTimerZero(true);
+      }, 1000);
+      return () => clearInterval(refreshIntervalId);
+    }
+  }, [gameTimer, gameTimerZero, headerGameCurrentTime]);
+
+  useEffect(()=>{ //update gameSession currentTimer every 3 seconds with time from headerGame Timer 
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+          let newUpdates = {currentTimer: (localStorage.getItem('currentGameTimeStor')>= 0 ? localStorage.getItem('currentGameTimeStor') : 0)};
+          apiClient.updateGameSession({ id: gameSessionId, ...newUpdates });
+      }, 3000);
+      return () => clearInterval(refreshIntervalId);
+    }
+  }, [gameTimer, gameTimerZero]);
+
   const handleUpdateGameSession = (newUpdates: Partial<IGameSession>) => {
     apiClient.updateGameSession({ id: gameSessionId, ...newUpdates })
       .then(response => {
+
+        if (response.currentState === stateArray[2])
```

Comment:
I'm not familiar with the host app, but this looks to be error-prune as if we change the `GameSessionState` order, it breaks it. Any reason for not using `GameSessionState` straight here? I'd highly recommend against this pattern as it's coupling the order of `GameSessionState` enum values to what they mean in the host app.

Reply:
Ok @maniramezan, I've updated all the comparisons to be like `nextState === GameSessionState.PHASE_1_DISCUSS`. I've removed the `stateArray` and used object.keys to determine the next state from the enum.